File: Site/Scripts/app/models/task.js
/// <reference path="../utilities.js"/>
/// <reference path="../constants.js"/>
define(['backbone', 'app/utilities', 'syncObject'], function (backbone, appUtilities, syncObject) {
	var task = backbone.Model.extend({
		// Default attributes for the task
		// and ensure that each task created has `title` and `completed` keys.
		defaults: {
			title: 'new task',
			description: '',
			children: [],
			status: "Action Pending",
			when: "Soon",
			where: "Work",
			lastPersisted: AppConstants.EndOfTime
		},
		initialize: function () {
			if (this.get("id") == null) {
				this.set("id", appUtilities.CreateGuid());
			}
			if (this.get("children") == null) {
				this.set("children", []);
			}
		},
		sync: syncObject.sync,
		//url: AppConstants.Url,
		isNew: function() {
		    return this.get("lastPersisted") == AppConstants.EndOfTime;
		}
	});
	return task;
});

File: WebApi/Models/Task.cs
using System;
using MongoDB.Bson.Serialization.Attributes;

namespace WebApi.Models
{
    [BsonIgnoreExtraElements]
	public class Task : IEquatable<Task>
	{
        [BsonElement("_id")]
		public string id { get; set; }
		public string title { get; set; }
		public string description { get; set; }
		public string status { get; set; }
		public string when { get; set; }
		public string where { get; set; }
		public string children { get; set; }
        public string lastPersisted { get; set; }
	    public bool Equals(Task other)
	    {
	        return other.id == id;
	    }
        public override bool Equals(object obj)
        {
            if (obj is Task)
            {
                return ((Task) obj).id == id;
            }
            return false;
        }
	}
}